kotest-runner/kotest-runner-junit5/src/jvmMain/kotlin/io/kotest/runner/junit/platform/KotestJunitPlatformTestEngine.kt

```diff
@@ -141,14 +139,99 @@ class KotestJunitPlatformTestEngine : TestEngine {
    }
 }
 
-class KotestEngineDescriptor(
+open class KotestEngineDescriptor(
    id: UniqueId,
    val classes: List<KClass<out Spec>>,
    val scripts: List<KClass<*>>,
    val testFilters: List<TestFilter>,
    val error: Throwable?, // an error during discovery
 ) : EngineDescriptor(id, "Kotest") {
+
+   private var isUniqueIdSelectorsRequest = false
+
    override fun mayRegisterTests(): Boolean = true
```

Comment:
Link to `isUniqueIdSelectorsRequest`, but we could probably get rid of this overridden `mayRegisterTests` if adopting a more JUnit idiomatic way

Reply:
This should still be true since Kotest registers additional test descriptors to report certain errors.